### ISSUE TITLE
PCC drop in YOLOS-Small model due to linear and sigmoid ops

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-xla#2313]
### Describe the bug

- YOLOS-Small having PCC=`0.853291928768158` . [nov25_yolos_small.log](https://github.com/user-attachments/files/23740475/nov25_yolos_small_org.log)
- This model returning [3 outputs](https://github.com/huggingface/transformers/blob/51f94ea06d19a6308c61bbb4dc97c40aabd12bad/src/transformers/models/yolos/modeling_yolos.py#L790C26-L790C56) and their respective PCC values are:
  - [logits](https://github.com/huggingface/transformers/blob/51f94ea06d19a6308c61bbb4dc97c40aabd12bad/src/transformers/models/yolos/modeling_yolos.py#L772) - 0.98780
  - [pred_boxes](https://github.com/huggingface/transformers/blob/51f94ea06d19a6308c61bbb4dc97c40aabd12bad/src/transformers/models/yolos/modeling_yolos.py#L773) - 0.85329
  - [Hidden_states](https://github.com/huggingface/transformers/blob/51f94ea06d19a6308c61bbb4dc97c40aabd12bad/src/transformers/models/yolos/modeling_yolos.py#L759C8-L759C28) - 0.99425
- This ticket involves discussion about PCC=0.85329 
- This drop originates from the combination of [linear](https://github.com/huggingface/transformers/blob/51f94ea06d19a6308c61bbb4dc97c40aabd12bad/src/transformers/models/yolos/modeling_yolos.py#L664C37-L664C46) + [sigmoid](https://github.com/huggingface/transformers/blob/51f94ea06d19a6308c61bbb4dc97c40aabd12bad/src/transformers/models/yolos/modeling_yolos.py#L773C58-L773C68) ops. 
- A sanity test including both ops together shows PCC = 0.8628583550453186.[nov25_sanity_1.log](https://github.com/user-attachments/files/23740676/nov25_sanity_1.log)
- When run individually with the saved inputs, both ops pass with valid pcc.
  - Linear - [nov25_sanity_2.log](https://github.com/user-attachments/files/23740621/nov25_sanity_2.log)
  - Sigmoid - [nov25_sanity_3.log](https://github.com/user-attachments/files/23740626/nov25_sanity_3.log)

### EmitPY Observations

- [linear_sigmoid.zip](https://github.com/user-attachments/files/23740964/linear_sigmoid.zip) ( Note : ttnn.sigmoid is not present)
- In sanity 1 tt-xla compilation log too, ttnn.sigmoid is absent but ttir.sigmoid is present [nov24_yolos_s1.log](https://github.com/user-attachments/files/23741025/nov24_yolos_s1.log)
- This suggests a missing ttir → ttnn sigmoid mapping, and therefore the absence of ttnn.sigmoid is likely the root cause.
- The TT-Metal experiments below confirm this conclusion.

### TT-Metal experiments

- Exp 1 : while running the TTNN repro generated by emitpy on metal , PCC =`0.8628582954406738` [nov25_linear_sigmoid_metal.log](https://github.com/user-attachments/files/23741116/nov25_linear_sigmoid_metal.log)
- Exp 2 : To confirm , absence of ttnn.sigmoid is the cause I removed sigmoid operation from cpu model & compared the results. PCC = `0.9999871253967285 `  [nov25_linear_sigmoid_metal_1.log](https://github.com/user-attachments/files/23741122/nov25_linear_sigmoid_metal_1.log)
- This confirms the conclusion

### TT-MLIR TICKET

- https://github.com/tenstorrent/tt-mlir/issues/6016

### Steps to reproduce the issue

In tt-xla, 

```
git checkout kkannan/nov24_yolos_debug
git submodule update --init --recursive
git lfs pull --include 'sig_slice_input_xla.pt l2_ip_xla.pt'

# Whole model run 
pytest tests/runner/test_models.py -k "yolos_small/pytorch-small-single_device-full-inference" -svv --runxfail

# sanity runs
# place the corresponding loader contents in yolos model's loader and rerun the above command 

# EmitPY
python examples/pytorch/codegen/yolos_sanity.py
```
In tt-metal ,

```
git checkout kkannan/nov25_linear_sigmoid_yolos
git submodule update --init --recursive
git lfs pull --include 'linear_sigmoid_ip.pt  tensors/*.tensorbin'

# exp1 
pytest tests/ttnn/unit_tests/operations/linear_sigmoid/test_linear_sigmoid.py -svv 

# exp2 
pytest tests/ttnn/unit_tests/operations/linear_sigmoid/test_linear.py -svv
```

### Needed Loader scripts

- [loader_sanity_1.py](https://github.com/user-attachments/files/23740720/loader.py)
- [loader_sanity_2.py](https://github.com/user-attachments/files/23740746/loader.py)
- [loader_sanity_3.py](https://github.com/user-attachments/files/23740755/loader.py)

**Comment by kamalrajkannan78** [comment]
Thanks for the fix @ddilbazTT 

On latest main, I got `flat_pccs=[tensor(0.98780), tensor(0.99027), tensor(0.99425)]` [dec2_yolos.log](https://github.com/user-attachments/files/23872881/dec2_yolos.log)
